rmc/utils/constraint.py

```diff
@@ -1969,6 +1973,120 @@ def _annotate_gene_ids(ht: hl.Table) -> hl.Table:
     )
 
 
+def check_for_overlapping_intervals(interval_ht: hl.Table, coord_ht: hl.Table) -> None:
+    """
+    Check for overlapping intervals within the same transcript.
+
+    :param interval_ht: RMC HT keyed by RMC interval.
+    :param coord_ht: RMC HT keyed by either RMC interval start or end coordinate.
+    :return: None
+    """
+    coord_ht = coord_ht.annotate(
+        interval_matches=interval_ht.index(coord_ht.locus, all_matches=True),
+    )
+    coord_ht = coord_ht.checkpoint(
+        f"{TEMP_PATH_WITH_FAST_DEL}/rmc_coord_check.ht", overwrite=True
+    )
+    coord_ht = coord_ht.filter(hl.len(coord_ht.interval_matches) > 1)
```

Comment:
Line probably redundant since you've got line 1998

Reply:
this may have been for efficiency -- I'll move it above the checkpoint